### DISCUSSION TITLE
Batch padding

### DISCUSSION THREAD

**Discussion by Spectra456** [speechbrain/speechbrain#1669] [Q&A] [answered]
Hi.
I'm using the pretrained ecapa-tdnn model as follows
`classifier = EncoderClassifier.from_hparams(source="speechbrain/spkrec-ecapa-voxceleb", savedir="pretrained_models/spkrec-ecapa-voxceleb")`
and trying to inference with some batches. Everything works fine, but samples which was padded is had little bit different embeddings. Is there any possible methods to avoid this?

**Comment by anautsch** [comment]
Hi @Spectra456 ok, please provide an example to clarify what was the expected result and what happened instead. 
(so we can better narrow down the problem)

**Comment by TParcollet** [comment, marked as answer]
This is related to the batch independent PR. It's in progress. A batch size of 1 will solve this issue, or simple use segments of equivalent lengths to avoid too much padding.

**Reply by danoneata** [reply]
Hello Titouan! Inconsistent results when batching seem to be an issue even today (speechbrain version 0.5.16). For example, the following code fails at the second assert (and even the first one needs a high tolerance to make it pass). Any ideas what causes this issue?

```python
import torch

from speechbrain.pretrained import EncoderClassifier
from speechbrain.dataio.batch import PaddedBatch

classifier = EncoderClassifier.from_hparams(source="speechbrain/spkrec-ecapa-voxceleb")

wav0 = torch.randn(32000)
wav1 = torch.randn(16000)

emb0 = classifier.encode_batch(wav0)
emb1 = classifier.encode_batch(wav1)

wavs = [{"wav": wav} for wav in [wav0, wav1]]
wavs = PaddedBatch(wavs)
embs = classifier.encode_batch(*wavs["wav"])

assert torch.allclose(emb0.squeeze(0), embs[0], atol=1e-04)
assert torch.allclose(emb1.squeeze(0), embs[1], atol=1e-04)
```

**Comment by TParcollet** [comment]
Hi, we abandoned this PR i believe as the difference shouldn't be too high while the refactoring necessary to achieve batch independancy was crazy big. @mravanelli can say more.

**Reply by danoneata** [reply]
I see, thanks a lot for your reply!
